### DISCUSSION TITLE
직렬화에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by 3Juhwan** [2024-woowacourse-study/level-interview#99] [Java]
.

**Comment by kunsanglee** [comment]
직렬화(Serialization)와 역직렬화(Deserialization)는 객체의 상태를 저장하고, 그 상태를 다시 객체로 복원하는 과정을 말합니다. 

직렬화는 객체의 상태를 바이트 스트림으로 변환하는 과정입니다. 이렇게 변환된 바이트 스트림은 파일 시스템에 저장하거나 네트워크를 통해 다른 시스템에 전송할 수 있습니다. 직렬화를 통해 객체의 데이터를 쉽게 교환하고, 애플리케이션 실행 상태를 영속화(persist)할 수 있습니다. 

역직렬화는 직렬화된 바이트 스트림을 다시 원래의 객체 상태로 복원하는 과정입니다. 파일 시스템에 저장되었거나 네트워크를 통해 전송된 바이트 스트림을 읽어들여, 그 내용으로 새로운 객체를 생성합니다. 

Java에서는 Serializable 인터페이스를 구현함으로써 객체를 직렬화 가능하게 할 수 있고, ObjectOutputStream과 ObjectInputStream 클래스를 통해 각각 직렬화와 역직렬화를 수행합니다. 이 과정은 데이터 교환 및 애플리케이션 상태의 영속화에 유용하지만, 보안과 호환성 문제에 주의해야 합니다.

**Comment by seunghye218** [comment]
직렬화 및 역직렬화는 객체를 바이트 스트림으로 변환한 다음 다시 객체로 재구성하는 과정을 다루는 개념입니다.

직렬화 및 역직렬화는 데이터 지속성, 네트워크 통신 등 다양한 Java 애플리케이션에서 널리 사용됩니다.

직렬화는 객체의 상태를 저장 또는 전송하고 나중에 재구성할 수 있는 바이트 스트림으로 변환하는 과정입니다. 이를 통해 개체를 파일에 저장하거나 네트워크를 통해 전송하거나 데이터베이스에 저장한 다음 나중에 재구성할 수 있습니다.

역직렬화는 직렬화의 반대 프로세스로, 바이트 스트림이 다시 객체로 변환됩니다. 바이트 스트림을 읽고 상태와 구조를 포함하여 객체를 재구성하는 작업이 포함됩니다.
